packages/web3-providers-ws/src/index.ts

```diff
@@ -98,7 +99,9 @@ export default class WebSocketProvider<
 		this._onCloseHandler = this._onClose.bind(this);
 
 		this._init();
-		this.connect();
+		if(autoConnect!==false){
+			this.connect();
```

Comment:
This is for first time connection , not for re connect so what is need of adding this check here?